tools/macos/README.md

```diff
@@ -1,8 +1,89 @@
-#docker-machine:
+# How to setup a Mac host for OpenWhisk
 
-* one time setup
+One way to develop or deploy OpenWhisk on a Mac is to use [docker-machine](https://docs.docker.com/machine/install-machine/) which runs the Docker daemon inside a virtual machine accessible from the Mac host.
+
+# Prerequisites
+
+The following are required to build and deploy OpenWhisk from a Mac host:
+
+- [Oracle VM VirtualBox](https://www.virtualbox.org/wiki/Downloads)
+- [Docker](https://docs.docker.com/engine/installation/mac/) (including `docker-machine`)
+- [Java 8](http://www.oracle.com/technetwork/java/javase/downloads/index.html)
+- [Scala 2.11](http://scala-lang.org/download/)
+- [Ant](http://ant.apache.org/manual/install.html)
+- [Ant-Contrib Tasks](http://ant-contrib.sourceforge.net/)
+- [Gradle](https://docs.gradle.org/current/userguide/installation.html)
+
+[Homebrew](http://brew.sh/) is an easy way to install all of these and prepare your Mac to build and deploy OpenWhik. The following shell command is provided for your convenience to install `brew` with [Cask](https://github.com/caskroom/homebrew-cask) and bootstraps these to complete the setup. Copy the entire section below and paste it into your terminal to run it.
+
+```
+echo '
+# install homebrew
+/usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
+# install cask
+brew tap caskroom/cask
+# install virtualbox
+brew cask install virtualbox
+# install docker
+brew install https://raw.githubusercontent.com/Homebrew/homebrew-core/e010e91a6bb5ad6697fb158017b36f575bbe644f/Formula/docker.rb
+# install docker-machine
+brew install docker-machine
+# install java 8, scala
+brew cask install java
+brew install scala
+# install ant and gradle
+brew install ant
+brew install ant-contrib
+brew install gradle
+# the following is required for running tests
+sudo pip install jsonschema' | bash
```

Comment:
I think we need `-H` if I don't use it I get a lot of warnings `sudo -H pip install jsonschema' | bash`